packages/stark-core/src/modules/http/helpers/http-helpers.ts

```diff
@@ -0,0 +1,45 @@
+import { HttpParameterCodec, HttpParams } from "@angular/common/http";
+import { StarkQueryParam } from "@nationalbankbelgium/stark-core";
+import { CustomEncoder } from "../entities/custom-encoder";
+
+/**
+ *  @ignore
+ */
+const _reduce: Function = require("lodash/reduce");
+
+/**
+ * A custom implementation of HttpParameterCodec to correctly encode query parameters.
+ * @link https://github.com/NationalBankBelgium/stark/issues/1130
+ */
+export const CUSTOM_ENCODER: HttpParameterCodec = new CustomEncoder();
+
+/**
+ * Converts the Map<string, StarkQueryParam> required by the service into a HttpParams object required by Angular
+ * @param starkQueryParam - params to convert
+ */
+export function convertStarkQueryParamsIntoHttpParams(starkQueryParam: Map<string, StarkQueryParam>): HttpParams {
+	return _reduce(
+		convertMapIntoObject(starkQueryParam), // convert to object
+		(httpParams: HttpParams, starkParams: StarkQueryParam, key: string) =>
+			starkParams
```

Comment:
Can you use `if` blocks here since we have nested blocks and it is more diffictul to read the code?

Reply:
I like actually really like this pattern. But maybe it's clearer like this?
```
typeof value === "undefined" // set key to empty string when not defined
	? httpParams.set(key, "")
	: Array.isArray(value) // append each string to the key when set to an array
	? _reduce(value, (acc: HttpParams, entry: string) => acc.append(key, entry), httpParams)
	: httpParams.set(key, value),
```